src/erp12/ga_clj/toolbox.cljc

```diff
@@ -29,42 +29,45 @@
 (defn lexicase-selection
   "Helper for make-lexicase-selection. Takes the population and cases shuffled
    in a random order. Behaves deterministically to aid in testing."
-  [candidates cases]
+  [candidates cases errors-key]
   ;; Stop when cases is empty or candidates has only 1 left.
   (if (or (empty? cases)
           (= 1 (count candidates)))
     (rand-nth candidates)
     (let [the-case (first cases)
-          best (apply min (map #(nth (:errors %) the-case)
+          best (apply min (map #(nth (get % errors-key) the-case)
```

Comment:
I think `error-key` should be treated as a function `(errors-key %)` in case the error vector isn't a top-level key.

Reply:
Good idea, changed.